src/main/java/io/github/dlinov/leetcode/Task207CourseSchedule.java

```diff
@@ -0,0 +1,40 @@
+package io.github.dlinov.leetcode;
+
+import java.util.*;
+
+public class Task207CourseSchedule {
+    class Solution {
+        public boolean canFinish(int numCourses, int[][] prerequisites) {
+            final Map<Integer, Set<Integer>> graph = new HashMap<>();
+            for (int i = 0; i < prerequisites.length; ++i) {
+                final int[] v = prerequisites[i];
+                final int target = v[0];
+                final int prereq = v[1];
+                graph.computeIfAbsent(target, x -> new HashSet<Integer>());
+                graph.get(target).add(prereq);
+            }
+            final Stack<Integer> stack = new Stack<>();
```

Comment:
Nice one! Usually topological sort is implemented in a recursive way ([1](https://github.com/linouk23/leetcode/blob/master/207.cpp), [2](https://e-maxx.ru/algo/topological_sort)).